frontend/src/components/pages/cabins/Admin/BookingSemesterPicker.tsx

```diff
@@ -0,0 +1,185 @@
+import { useMutation } from "@apollo/client";
+import ErrorDialog from "@components/dialogs/ErrorDialog";
+import { UPDATE_BOOKING_SEMESTER } from "@graphql/cabins/mutations";
+
+import useBookingSemester from "@hooks/cabins/useBookingSemester";
+import {
+  Box,
+  Button,
+  Checkbox,
+  CircularProgress,
+  FormControlLabel,
+  Grid,
+  Snackbar,
+  TextField,
+  Tooltip,
+  Typography,
+} from "@material-ui/core";
+
+import { Alert } from "@material-ui/lab";
+import dayjs from "dayjs";
+import router from "next/router";
+import React, { useState } from "react";
+
+export type BookingSemester = {
+  fallStartDate: string;
+  fallEndDate: string;
+  springStartDate: string;
+  springEndDate: string;
+  fallSemesterActive: boolean;
+  springSemesterActive: boolean;
+};
+
+export const defaultBookingSemester: BookingSemester = {
+  fallStartDate: "",
+  fallEndDate: "",
+  springStartDate: "",
+  springEndDate: "",
+  fallSemesterActive: false,
+  springSemesterActive: false,
+};
+
+const BookingSemesterPicker = () => {
+  const [updateBookingSemester] = useMutation<{ semesterData: BookingSemester }>(UPDATE_BOOKING_SEMESTER);
+  const handleErrorDialogClose = () => router.push("/");
+  const [openSnackbar, setOpenSnackbar] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState("");
+  const [alertSeverity, setAlertSeverity] = useState<"success" | "error">("success");
+
+  const { bookingSemester, setBookingSemester, loading, error } = useBookingSemester();
+
+  const validateBookingSemesters = (): boolean => {
+    const fallStart = dayjs(bookingSemester.fallStartDate);
+    const fallEnd = dayjs(bookingSemester.fallEndDate);
+    const springStart = dayjs(bookingSemester.springStartDate);
+    const springEnd = dayjs(bookingSemester.springEndDate);
+    if (fallStart.isAfter(fallEnd) || springStart.isAfter(springEnd) || fallEnd.isAfter(springStart)) {
+      setSnackbarMessage("Start-datoer kan ikke vært før slutt-datoer.");
+      setOpenSnackbar(true);
+      setAlertSeverity("error");
+      return false;
+    }
+    return true;
+  };
+
+  const handleUpdate = () => {
+    if (validateBookingSemesters()) {
+      updateBookingSemester({ variables: { semesterData: { ...bookingSemester } } }).then(() => {
+        setAlertSeverity("success");
+        setSnackbarMessage("Booking-semestrene er nå oppdatert.");
+        setOpenSnackbar(true);
+      });
+    }
+  };
+
+  return (
+    <Box>
+      {loading ? (
+        <CircularProgress />
+      ) : (
+        <>
+          <ErrorDialog error={error} handleErrorDialogClose={handleErrorDialogClose} />
```

Comment:
On the topic of patterns and NextJS, here we also have some potential for making some improvements to our site! Introducing 🥁 [dynamic imports](https://nextjs.org/docs/advanced-features/dynamic-import#basic-usage) 🎉